src/Common/src/CoreLib/System/Memory.cs

```diff
@@ -166,9 +174,9 @@ public Memory<T> Slice(int start, int length)
         {
             if ((uint)start > (uint)_length || (uint)length > (uint)(_length - start))
             {
-                ThrowHelper.ThrowArgumentOutOfRangeException();
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.start);
```

Comment:
The argument name is not accurate - it can be length that is wrong. I think we should not bother with supplying the argument name here.

Reply:
this is the code I picked from the System.Memory in corefx but I agree with you. I'll fix that.